inc/compatibility/elementor.php

```diff
@@ -456,6 +456,24 @@ public static function is_elementor_template( $elementor_template_type ) {
 		return self::$cache_cp_has_template[ $location ];
 	}
 
+	/**
+	 * Detect if the a page is using the checkout widget.
+	 */
+	public static function is_elementor_checkout() {
+		if ( ! class_exists( '\ElementorPro\Plugin', false ) ) {
+			return false;
+		}
+
```

Comment:
We can add a `is_checkout()` check here (after the if Elementor is active check). That helps to restrict the execution of the method only on WC checkout page. Otherwise, that method will be executed for all pages, and cause unnecessary costs even though it's a cheap operation.

I assume if a user used Elementor checkout widget, the user matched the page as the WC checkout endpoint. At least, most of them, I think.